runners/google-cloud-dataflow-java/worker/src/main/java/org/apache/beam/runners/dataflow/worker/DataflowMetricsContainer.java

```diff
@@ -29,7 +30,11 @@
  * An implementation of {@link MetricsContainer} that reads the current execution state (tracked in
  * a field) to determine the current step. This allows the {@link MetricsEnvironment} to only be
  * updated once on entry to the entire stage, rather than in between every step.
+ *
+ * <p><b>Not actually serializable</b>. {@link MetricsContainer} is Serializable only due to its
```

Comment:
That's a strange behavior. Could you add this javadoc comment to the `MetricsContainer` base class? Seems like we should eventually remove `Serializable` and ensure it doesn't end up within a serialized closure.